teradata/datadog_checks/teradata/check.py

```diff
@@ -0,0 +1,186 @@
+            cursor.execute(query)
+            if cursor.rowcount < 1:
+                self._query_errors += 1
+                self.log.warning('Failed to fetch records from query: `%s`.', query)
+                return None
+            for row in cursor.fetchall():
+                query_name = re.search(r'(DBC.[^\s]+)', query).group(1)
+                try:
+                    yield self._queries_processor(row, query_name)
+                except Exception as e:
+                    self.log.debug(
+                        'Unable to process row returned from query "%s", skipping row %s. %s', query_name, row, e
+                    )
+                    yield row
+
+    def _executor_error_handler(self, error):
+        # type: (AnyStr) -> AnyStr
+        self._query_errors += 1
+        return error
+
+    @contextmanager
+    def connect(self):
+        # type: () -> Iterator[teradatasql.connection]
+        conn = None
+        if TERADATASQL_IMPORT_ERROR:
+            self.log.error(
+                'Teradata SQL Driver module is unavailable. Please double check your installation and refer to the '
+                'Datadog documentation for more information. %s',
+                TERADATASQL_IMPORT_ERROR,
+            )
+            raise TERADATASQL_IMPORT_ERROR
+        self.log.info('Connecting to Teradata database %s on server %s.', self.config.database, self.config.server)
+        try:
+            conn = teradatasql.connect(self._connect_params)
+            self.log.info('Connected to Teradata.')
+            yield conn
+        except Exception as e:
+            self.log.error('Unable to connect to Teradata. %s.', e)
+            raise e
+        finally:
+            if conn:
+                conn.close()
+
+    def submit_health_checks(self):
+        # type: () -> None
+        connect_status = ServiceCheck.OK
+        query_status = ServiceCheck.CRITICAL if self._query_errors else ServiceCheck.OK
+
+        self.service_check(SERVICE_CHECK_QUERY, query_status, tags=self._tags)
+        self.service_check(SERVICE_CHECK_CONNECT, connect_status, tags=self._tags)
+
+    def _queries_processor(self, row, query_name):
+        # type: (Sequence, AnyStr) -> Sequence
+        """
+        Validate timestamps, filter tables, and normalize empty tags.
+        """
+        unprocessed_row = row
+
+        # Return database version immediately
+        if query_name == 'DBC.DBCInfoV':
+            submit_version(self, row)
+            return unprocessed_row
+
+        # Only Resource Usage rows include timestamps and also do not include tags.
+        if query_name == 'DBC.ResSpmaView':
+            processed_row = timestamp_validator(self, unprocessed_row)
+            return processed_row
+
+        # Only AllSpaceV rows include table tags
+        if (
+            query_name == 'DBC.AllSpaceV'
+            and is_affirmative(self.config.collect_table_disk_metrics)
+            and self._tables_filter
```

Comment:
Where else do you use `_tables_filter`? It looks like you call `self._tables_filter = create_tables_filter(self)` on line 98, which returns you a tuple of `(tables_to_collect, tables_to_exclude)`, although `_tables_filter` is only used in this if statement? Do the `tables_to_collect` and `tables_to_exclude` get used elsewhere?

Reply:
They're used [here](https://github.com/DataDog/integrations-core/pull/11701/files#diff-38f6a8d8c626e687bc982cf07826b7435ab47a9a09b48d988f67b932f96924f0R12) in the `filter_tables` utils method, which filters the row by table after the row has been collected.